packages/react-is/src/ReactIs.js

```diff
@@ -56,6 +56,31 @@ export function typeOf(object: any) {
   return undefined;
 }
 
+export function elementType(type: any) {
+  switch (type) {
+    case REACT_ASYNC_MODE_TYPE:
+    case REACT_FRAGMENT_TYPE:
+    case REACT_PROFILER_TYPE:
+    case REACT_STRICT_MODE_TYPE:
+      return type;
+  }
+  const typeofType = typeof type;
+  switch (typeofType) {
+    case 'string':
+    case 'function':
+      return REACT_ELEMENT_TYPE;
+    case 'object':
```

Comment:
It is very likely that we're going to do a breaking change and actually turn these into just functions.